app/server/lib/Triggers.ts

```diff
@@ -497,6 +496,9 @@ export class DocTriggers {
     tableDataAction: TableDataAction,
   ) {
     const bulkColValues = fromTableDataAction(tableDataAction);
+    // HACK: bulkColValues don't include removed data because the data no longer exist on the Database
+    // but tableDataAction is got from Database query
+    this._appendRemovedRowsIntoTableColValues(tableDelta, bulkColValues);
```

Comment:
If I understand your intent, you would like the `remove` event to give the data of the deleted record, right?

I am not sure this would be necessary.
In comparison, DELETE operations in REST normally don't pass any body to the request ([not allowed by the browsers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Methods/DELETE)), but the ID through the request path.

As you are in the context of the webhook, I would just pass the ID (if I remember correctly in the body) and nothing else.

What do you think?

Reply:
hello. I've just casually stumbled into this PR out of curiosity because I would need it for my own use case. For what I need, the body would be really useful and the comparison with REST doesnt' stand, I think.

In REST, you can request the resource to know the content and then deleted it. With this trigger, the resource will be gone once you send the ID to the webhook and you don't know what was there.

In my specific case, I use an email as a primary key throughout the system and if a row is deleted in grist, I would need to delete it elsewhere in the system. The webhook would be very useful, but it should send me the email to know what to do with this info. Only the ID would be meaningless.

I hope this comment was useful and not off-topic.